MoviesMVVM/MoviesMVVM/WebViewController.swift

```diff
@@ -52,19 +51,24 @@ final class WebViewController: UIViewController {
 
     private func loadWebViewData() {
         guard let index = filmIndex else { return }
-        Service.shared.loadVideos(index: index) { [weak self] result in
-
-            guard result.count != 0,
-                  let url =
-                  URL(string: "\(Constants.urlShemeHostPath)\(index)\(Constants.urlFragment)\(result[0].key)")
-            else {
+        networkService.loadVideos(index: index) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .success(result):
+                guard result.count != 0,
+                      let url =
+                      URL(string: "\(Constants.urlShemeHostPath)\(index)\(Constants.urlFragment)\(result[0].key)")
+                else {
+                    DispatchQueue.main.async {
+                        self.dismiss(animated: true)
+                    }
+                    return
+                }
                 DispatchQueue.main.async {
-                    self?.dismiss(animated: true)
+                    self.webView.load(URLRequest(url: url))
                 }
-                return
-            }
-            DispatchQueue.main.async {
-                self?.webView.load(URLRequest(url: url))
+            case let .failure(error):
+                self.showErrorAlert(title: Constants.errorString, message: error.localizedDescription)
```

Comment:
```suggestion
                self.showErrorAlert(title: Constants.errorString, message: error.localizedDescription)
```
вот тут вопрос, showErrorAlert это же UI, возможно и тут нужен DispatchQueue.main.async